sds.c

```diff
@@ -1018,7 +1018,14 @@ sds *sdssplitargs(const char *line, int *argc) {
                 if (*p) p++;
             }
             /* add the token to the vector */
-            vector = s_realloc(vector,((*argc)+1)*sizeof(char*));
+            char **data = s_realloc(vector,((*argc)+1)*sizeof(char*));
+            if (data == NULL) {
+                /* Return something not NULL. */
+                return s_malloc(sizeof(void*));
```

Comment:
I agree with what you're saying. I had put this as part of my patch because hiredis does it https://github.com/antirez/redis/blob/unstable/deps/hiredis/sds.c#L1027

I can certainly change this. I just don't have enough context to assume what the maintainers would want.